Lib/test/_test_multiprocessing.py

```diff
@@ -1056,6 +1056,19 @@ def __reduce__(self):
             self.assertTrue(q.get(timeout=1.0))
             close_queue(q)
 
+        with test.support.captured_stderr():
+            # bpo-33078: verify that the queue size is correctly handled
+            # on errors.
+            q = self.Queue(maxsize=1)
+            q.put(NotSerializable())
+            q.put(True)
+            self.assertEqual(q.qsize(), 1)
```

Comment:
Is this check correct? Depending on when the worker thread triggers, it might return 2, no?

Reply:
I think it is correct because `maxsize=1` so the second `q.put` should return only once the failure has been triggered.

Reply:
That's a good point, thank you.